hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/AbstractEmbeddableMapping.java

```diff
@@ -91,8 +92,12 @@ public Object[] getValues(Object compositeInstance) {
 			return optimizer.getAccessOptimizer().getPropertyValues( compositeInstance );
 		}
 
+		return getAttributeValues( compositeInstance );
```

Comment:
What do you think about adding the class name as string as last value to this array if the type is polymorphic, similar to what happens for `CallableStatement` handling?